meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-configs-rpi-enable-mender-requirements.patch

```diff
@@ -55,11 +58,14 @@ index 4e8204ef88..4f4029e006 100644
  CONFIG_SYS_RELOC_GD_ENV_ADDR=y
  CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
  CONFIG_DM_KEYBOARD=y
-@@ -43,3 +41,4 @@ CONFIG_SYS_WHITE_ON_BLACK=y
+@@ -43,3 +41,7 @@ CONFIG_SYS_WHITE_ON_BLACK=y
  CONFIG_CONSOLE_SCROLL_LINES=10
  CONFIG_PHYS_TO_BUS=y
  CONFIG_OF_LIBFDT_OVERLAY=y
 +CONFIG_ENV_IS_IN_MMC=y
++CONFIG_ENV_OFFSET=0x400000
++CONFIG_ENV_OFFSET_REDUND=0x800000
++CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
```

Comment:
Would it be worth it to provide these in [u-boot-mender-common.inc](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc) in a post configure step based on `MENDER_PARTITION_ALIGNMENT`. This would be similar to what we have been doing up until now.

Or do you see any big drawbacks with attempting above?

Reply:
Yes, it annoys me a bit that something that was automatic before, won't be anymore. But patching this is much more intrusive than before, since we can no longer introduce a new file, we have to patch it inline. This is part of the auto-patcher, so my thinking was that maybe we should keep the non-auto-patching path clean?

What do you think?

Reply:
I am torn.  Agree with you, annoying and we should keep the non-auto-patching path clean. 

But this will also break all the existing board integrations using non-auto-patching. I guess the middle ground is providing more explicit information in the error message as I would expect many people to hit this when they upgrade from warrior -> zeus.

Reply:
Yes, I'm also torn. 🙄 But I think I agree that it's better to keep the non-auto-patching clean. I guess the good news is that moving it into Kconfig increases the chance that the auto-patcher will work.

But good point about the message, I'll try to make it more explicit, and mention exactly what people need to do to fix it.

Reply:
Looks like this needs a bit more work. Not defining the parameters actually crash the *build machine*! I have filed it under [MEN-3476](https://tracker.mender.io/browse/MEN-3476), but will return to it soon.

Reply:
@mirzak: Take a look now! I have added links to our Troubleshooting section in the error messages in the patch, and this goes together with https://github.com/mendersoftware/mender-docs/pull/952.

Reply:
Very nice information on what needs to be done to remedy the problems. Thank you for improving this and this patch set looks good to me